build.gradle

```diff
@@ -48,7 +48,7 @@ task clean(type: Delete) {
 }
 
 ext {
-    awsSdkVersion = '2.16.5'
+    awsSdkVersion = '2.16.6'
```

Comment:
Perhaps we should use `2.16.+` so that each minor version update doesn't break our compiler?

Reply:
The build failure is caused by the [`NewerVersionsAvailable`](https://android.googlesource.com/platform/tools/base/+/24973b0/lint/libs/lint-checks/src/main/java/com/android/tools/lint/checks/GradleDetector.java#210) lint. It would ordinarily be just a warning, but we have warnings [configured as errors](https://github.com/aws-amplify/amplify-android/blob/master/configuration/android-lint.gradle#L18). Projects that don't have this configuration tend to have lots of unaddressed warnings. All in all, I think this is the right setting.

I think there are ~three options here, and each has pros and cons:
1. The default, use `2.16.6`, and leave the `NewerVersionsAvailable` lint as a warning. Almost certainly, the warning will go un-addressed.
2. Use `2.16.6`, but make the `NewerVersionsAvailable` an error, as it is now. With this, we always know the exact version being used, and that it is the latest. But, sometimes our build randomly breaks.
3. Use `2.16.+`. With this, the build always works, and it always used the latest. But, it is harder to tell what version was used for any particular build. Things change somewhat "silently", under the hood. Of course, that's happens when a dependency publishes a new version without updating the point release, which is ~common. So, all this would really say is "AWS SDK point releases are not a big deal."

TLDR I think I like the `2.16.+`, too.

Reply:
(There was a thread of discussion around this, I think I might have accidentally deleted it.)

TLDR sounds good, let's do that in another PR